crates/runtime/src/spice_metrics.rs

```diff
@@ -32,38 +39,58 @@ pub enum Error {
     #[snafu(display("Error queriing prometheus metrics: {source}"))]
     FailedToQueryPrometheusMetrics { source: reqwest::Error },
 
-    #[snafu(display("Error getting timestamp: {source}"))]
-    UnableToGetTimestamp { source: std::time::SystemTimeError },
-
-    #[snafu(display("Error converting timestamp to seconds"))]
-    UnableToConvertTimestampToSeconds {},
-
     #[snafu(display("Error parsing prometheus metrics: {source}"))]
     UnableToParsePrometheusMetrics { source: std::io::Error },
+
+    #[snafu(display("Error writing to metrics table: {source}"))]
+    UnableToWriteToMetricsTable { source: internal_table::Error },
+
+    #[snafu(display("Error creating metrics table: {source}"))]
+    UnableToCreateMetricsTable { source: internal_table::Error },
 }
 
 pub struct MetricsRecorder {
     socket_addr: Arc<SocketAddr>,
-    table: Arc<MetricsTable>,
+    metrics_table: Arc<InternalTable>,
 }
 
 impl MetricsRecorder {
-    pub fn table(&self) -> Arc<MetricsTable> {
-        Arc::clone(&self.table)
+    pub fn metrics_table(&self) -> Arc<InternalTable> {
+        Arc::clone(&self.metrics_table)
     }
 }
 
 impl MetricsRecorder {
-    pub fn new(socket_addr: SocketAddr) -> Self {
-        let table = MetricsTable::new();
-
-        Self {
+    pub async fn new(socket_addr: SocketAddr) -> Result<Self, Error> {
+        let retention = Retention::new(
+            Some("timestamp".to_string()),
+            Some(TimeFormat::UnixSeconds),
+            Some(Duration::from_secs(1800)), // delete metrics older then 30 minutes
+            Some(Duration::from_secs(300)),  // run retention every 5 minutes
+            true,
+        );
+
+        let internal_table = InternalTable::new(
+            "metrics",
+            get_metrics_schema(),
+            Acceleration::default(),
+            Refresh::default(),
+            retention,
+        )
+        .await
+        .context(UnableToCreateMetricsTableSnafu)?;
+
+        Ok(Self {
             socket_addr: Arc::new(socket_addr),
-            table: Arc::new(table),
-        }
+            metrics_table: Arc::new(internal_table),
+        })
     }
 
-    async fn tick(socket_addr: &SocketAddr, table: &Arc<MetricsTable>) -> Result<(), Error> {
+    async fn tick(
+        socket_addr: &SocketAddr,
+        data_fusion: &Arc<RwLock<DataFusion>>,
```

Comment:
```suggestion
        datafusion: &Arc<RwLock<DataFusion>>,
```

datafusion is one word